Mathlib/GroupTheory/Submonoid/Membership.lean

```diff
@@ -412,6 +412,18 @@ theorem exists_multiset_of_mem_closure {M : Type*} [CommMonoid M] {s : Set M} {x
 #align submonoid.exists_multiset_of_mem_closure Submonoid.exists_multiset_of_mem_closure
 #align add_submonoid.exists_multiset_of_mem_closure AddSubmonoid.exists_multiset_of_mem_closure
 
+@[to_additive]
+theorem mem_closure_of_exists_multiset {M : Type*} [CommMonoid M] {s : Set M} {x : M}
```

Comment:
This is a legit proof, but I had the impression that in mathlib, people tried to avoid explicit lemmas like this in benefit of induction lemmas. (Avoiding them would require to define the induction lemma for “divisor_closure”.)

Reply:
I have added the induction lemma.